content/post/adding_context_with_annotations.md

```diff
@@ -0,0 +1,106 @@
+
+In this post I will walk through the steps I take to integrate annotations into my data models and visualizations. I will refer to an imaginary home goods e-commerce company called “Utensell” to guide the walkthrough.
+
+### Step 1: Whenever something notable happens, record it in a google spreadsheet. Easy!
+
+[screenshot_1]: https://github.com/locallyoptimistic/LocallyOptimistic/blob/master/static/img/annotations-google-sheet.jpg
+
+Examples of things I keep note of:
+* Product Launches (e.g., launched new silverware sku)
+* Marketing Promotions - both online and offline (e.g., pulled back on marketing spend temporarily)
+* Feature Releases (e.g., Utensell introduced filtering functionality to our products page)
+* Press (e.g., Utensell founders featured on the How I Built This podcast)
+* Onsite Bugs (e.g., new users were not able to sign up for the Utensell newsletter)
+* Analytics Bugs (e.g., accidentally stopped capturing clicks on the “remove item” button in checkout)
+* Competitors Behavior (e.g., Amazon prime day)
+* Fraud (e.g., surge in bot traffic)
+
+Depending on the business, it may make sense to keep note of some additional metadata. In this example, I have a boolean flag to indicate whether there was a discount promotion associated with the event. I also keep track of links that are relevant to the notable event (e.g., links to press articles) in a separate column.
+
+_Daily Time Spent: I generally spend 1-5 minutes per day on this step._
+
+
+### Step 2: Automate the formatting and uploading of your annotations log to your database.
+
+I use the [datasheets](https://github.com/Squarespace/datasheets) python library (created by the team at Squarespace) to reformat and upload my annotations log table into Utensell’s data warehouse. But you can use any method you prefer! [Quick shoutout to Rujuta Kortikar, who built V1 of this script using the gspread library before datasheets was released].
+
+[screenshot_2]: https://github.com/locallyoptimistic/LocallyOptimistic/blob/master/static/img/annotations-jupyter-notebook.jpg
+
+_Daily Time Spent: This script runs automatically in Airflow so I don’t spend any time on this step anymore but the initial setup took around 3 hours._
+
+### Step 3: Reference your annotations log whenever and wherever you’d like!
+
+Let’s imagine that I want to run an analysis that looks at daily orders data for the past 7 days. Knowing that discount promotions lead to big spikes in order volume, I may want to incorporate that information into my analysis. One way to achieve this would be to join my annotations log table to my order facts table. Example below -
+
+```
+SELECT
+ orders.created_at AS order_created_date,
+ CASE WHEN annotations.discount_promotion_flag IS TRUE
+   THEN 'Discount Promotion' ELSE 'No Promotion' END AS discount_promotion_flag,
+ count(DISTINCT orders.id) AS total_orders
```

Comment:
This is such a bike-shed comment, but _my_ preferred SQL formatting syntax for this statement would look like:

```
CASE 
   WHEN annotations.discount_promotion_flag IS TRUE
       THEN 'Discount Promotion' 
   ELSE 'No Promotion' 
END AS discount_promotion_flag,
COUNT(DISTINCT orders.id) AS total_orders
```

But this is a Michael style rule, not a LO style rule so feel free to leave as is :)